src/NUnitFramework/framework/Constraints/ExceptionTypeConstraint.cs

```diff
@@ -37,24 +37,47 @@ public class ExceptionTypeConstraint : ExactTypeConstraint
         /// </summary>
         public ExceptionTypeConstraint(Type type) : base(type) { }
 
-        // TODO: This needs tests. May need a special result type.
-        ///// <summary>
-        ///// Write the actual value for a failing constraint test to a
-        ///// MessageWriter. Overriden to write additional information 
-        ///// in the case of an Exception.
-        ///// </summary>
-        ///// <param name="writer">The MessageWriter to use</param>
-        //public override void WriteActualValueTo(MessageWriter writer)
-        //{
-        //    Exception ex = actual as Exception;
-        //    base.WriteActualValueTo(writer);
-
-        //    if (ex != null)
-        //    {
-        //        writer.WriteLine(" ({0})", ex.Message);
-        //        writer.Write(ex.StackTrace);
-        //    }
-        //}
+        /// <summary>
+        /// Applies the constraint to an actual value, returning a ConstraintResult.
+        /// </summary>
+        /// <param name="actual">The value to be tested</param>
+        /// <returns>A ConstraintResult</returns>
+        public override ConstraintResult ApplyTo<TActual>(TActual actual)
+        {
+            actualType = actual == null ? null : actual.GetType();
```

Comment:
What if actualType is not an Exception? Take a look at string constraints, for example, to see what we do when a constraint that expects a certain kind of actual value doesn't get it. Admittedly, it's a highly unlikely scenario requiring explciit construction of the constraint.